### DISCUSSION TITLE
Configuration management

### DISCUSSION THREAD

**Discussion by zerkms** [vimeo/psalm#11057] [Q&A]
I have many (like > 50) local php projects, each of which uses psalm for static analysis.

It is annoying and error prone to go and change `psalm.xml` config contents every time I figure out some new parameter need to be added, or changed.

Question: how everybody handles it?

From what I found there is no really a way to inherit psalm configs. And psalm only accepts `.xml` as a config format (if it was for instance `.php` it would be possible to have a shared package with config and just include it everywhere).

It feels like I need to implement a plugin that changes the configuration in runtime?

Thank you!

**Comment by weirdan** [comment]
Psalm supports XInclude in its configs. See https://psalm.dev/docs/running_psalm/configuration/. That probably won't help with root tag attributes though.
